postgres-appliance/postgres_backup.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+function wait_for_cluster
+{
+    local SECONDS=${1}
+    [[ -z $1 ]] && SECONDS=0
+    local FINAL_EPOCH=$(date --date "$SECONDS seconds" +%s)
+
+    while true
+    do
+        pg_isready --quiet && return 0
+        [[ $(date +%s) -lt ${FINAL_EPOCH} ]] || break
+        sleep 5
+    done
+
+    echo "ERROR: cluster was not available after $SECONDS seconds" >&2
+    return 1
+}
+
+[[ -z $2 ]] && echo "Usage: $0 WALE_ENV_DIR PGDATA" && exit 1
+
+if [[ ! -z ${INITIAL_BACKUP+1} ]]
```

Comment:
Variable expansion, I just want to know whether the variable is set or not, I don't care about the value, see:
http://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash

```bash
$ export TEST=
$ [[ ! -z ${TEST} ]] && echo "Yes"
$ [[ ! -z ${TEST+1} ]] && echo "Yes"
Yes
$ unset TEST
$ [[ ! -z ${TEST+1} ]] && echo "Yes"
$
```